pydis_site/apps/content/resources/guides/pydis-guides/contributing/bot.md

```diff
@@ -2,207 +2,643 @@
+* `docker-compose up redis` to start the Redis container. You only need this if you're not using fakeredis. For more info refer to [Working with Redis](#optional-working-with-redis).
+
+You can start several services together: `docker-compose up web snekbox redis`.
+
+##### Setting Up a Development Environment
+The bot's code is Python code like any other. To run it locally, you will need the right version of Python with the necessary packages installed:
+
+1. Make sure you have [Python 3.9](https://www.python.org/downloads/) installed. It helps if it is your system's default Python version.
+2. Install Poetry by following [these instructions](https://github.com/python-poetry/poetry#installation), or with `pip install poetry`. In case of the latter, make sure pip installs to the right Python version.
+3. [Install the dependencies](../installing-project-dependencies).
+
+Assuming you have at least the site running in Docker already, you can now start the bot locally through the command line, or through your preferred IDE. Notice that the bot is started as a module.
+<div class="card">
+    <button type="button" class="card-header collapsible">
+        <span class="card-header-title subtitle is-6 my-2 ml-2">Ways to run code</span>
+        <span class="card-header-icon">
+            <i class="fas fa-angle-down title is-5" aria-hidden="true"></i>
+        </span>
+    </button>
+    <div class="collapsible-content">
+        <div class="card-content">
+            There are several ways to start the bot:
+            <ul>
+                <li>Through the command line, inside the bot directory, with either <code>poetry run task start</code>, or directly <code>python -m bot</code>.</li>
+                <li>If using PyCharm, enter <code>Edit Configurations</code> and set everything according to this image: <img src="/static/images/content/contributing/pycharm_run_module.png"></li>
+                <li>If using Visual Studio Code, set the interpreter to the poetry environment you created. In <code>launch.json</code> create a new Python configuration, and set the name of the program to be run to <code>bot</code>. VSC will correctly run it as a module.</li>
+            </ul>
+        </div>
+    </div>
+</div>
+<br>
+
+#### With More Things Running Locally
+You can run additional services on the host. For the site, refer to the [site contributing guide](../site) to learn how to start it on the host, in which case you will need to change `urls.site` in `config.yml` to wherever the site is being hosted. This guide won't go over how to install and start the other dependencies on the host. If possible, prefer to start the services through Docker to replicate the production environment as much as possible.
+
+#### Start Only the Bot with Docker
+This method will start only the bot using Docker. The site has to have been started somehow beforehand.
```

Comment:
Like having the site run in a separate docker instance? What other options are there?
This feels odd enough to not include without further explanation.